src/resources/filters/layout/typst.lua

```diff
@@ -121,7 +121,18 @@ end, function(layout)
     end)
   end)
   cells:insert(pandoc.RawInline("typst", ")\n"))
-  if layout.float.has_subfloats then
+  local has_subfloats = layout.float.has_subfloats
+  -- check for id-less figures that weren't made FloatRefTarget
+  if not has_subfloats then
+    _quarto.ast.walk(layout.float.content, {
+      Figure = function(figure)
+        if figure.id == nil or figure.id == "" then
```

Comment:
Do we even need this check?

In theory, we only see a `Figure` inside a `FloatRefTarget` if its id doesn't match the `<REF_TYPE>-<name>` pattern. But a "badly-formed id" should still not cause this rendering bug if we could avoid it.

I wouldn't go out of my way to _fix_ the case where the ids are badly formed, but unless I'm missing something (totally possible!), I don't see what this check is doing.

Reply:
Yes, even more so since it is `identifier`, not `id`, so this was always passing. 😞 

It would indeed fail

````
::: {#fig-panel2 layout-ncol=2}

![Placeholder]({{< placeholder 200 >}}){#foo}

![Placeholder]({{< placeholder 200 >}}){#bar}

Panels

:::
````

if written `figure.identifier`

Adding this test.

Reply:
`id` isn't an alias for `identifier` in a `Figure`? Huh. I'm ... fairly sure it is in `Span` and `Div`.